### PR TITLE
STYLE use absolute imports in asv_bench

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

No longer limiting `abs-imports` to just pandas. This requires telling `isort` that `benchmarks` is a local folder, else it groups it with third-party imports

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -202,7 +202,7 @@ repos:
     hooks:
     -   id: no-string-hints
 -   repo: https://github.com/MarcoGorelli/abs-imports
-    rev: v0.1.2
+    rev: v0.1.3
     hooks:
     -   id: abs-imports
-        files: ^pandas/
+        args: [--application-directories, '.,./asv_bench']
```
Comment (jreback) [inline .pre-commit-config.yaml:208]: what are the commas?

Comment (MarcoGorelli) [reply .pre-commit-config.yaml:208]: application directories, comma-separated. So, the first one is `.` (which contains `pandas`), and the second one is `./asv_bench` (which contains `benchmarks`)

**Comment by jorisvandenbossche** [comment]
IMO this is not an improvement in code clarity. The current `from .pandas_vb_common import` makes it very explicit that it is from a local file. When changing this to `from benchmarks...`, it might seem that this is some existing `benchmarks` package that is being imported.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jreback) [reply .pre-commit-config.yaml:208]: maybe its the quotes as you have a comma *inside* the 2nd one, this is really odd. but ok

**Comment by MarcoGorelli** [comment]
> IMO this is not an improvement in code clarity. The current `from .pandas_vb_common import` makes it very explicit that it is from a local file. When changing this to `from benchmarks...`, it might seem that this is some existing `benchmarks` package that is being imported.

that's a good point - Isn't it clear from the sections though? As in, if there is

```python

import numpy as np

import pandas as pd

from benchmarks.pandas_vb_common import tm
```
then it's clear that `benchmarks` is a local folder, because it appears after `pandas` and wasn't grouped with the `numpy` import.

No objection to closing though if you prefer these to be relative imports

File: .pre-commit-config.yaml (hunk 1 above)
Comment (MarcoGorelli) [reply .pre-commit-config.yaml:208]: yes, cause otherwise yaml doesn't parse it correctly. maybe a semi-colon would be a better separator

**Comment by jorisvandenbossche** [comment]
> Isn't it clear from the sections though?

But then you need to know our ordering rules by heart to infer it from that (personally I never care about ordering and just let isort fix it). I know that for code within the pandas package the pandas imports always come last (so that's an easy rule to remember), but in this case 1) it are not imports from pandas, but from some `benchmarks` package and 2) personally for other code I almost always import more specialized packages after numpy and pandas, so this is a rather general pattern I think contributors might be used to (at least I am). 

For me the main question is: what's the benefit? Does it make code more readable, or does it help automating style issues? (But did we ever have a problem with a person doing relative imports in the benchmarls that we didn't want and thus an automatic style check would be hepful?)

(and to be clear, I am very much in favor *generally* of automatic checks for certain things (and thanks a lot for your work on that), but we should still always question the reason for adding one)

**Comment by MarcoGorelli** [comment]
Thanks @jorisvandenbossche , that's a good point

> what's the benefit? 

I was just thinking that it would be clearer to have a rule applied everywhere rather than having `files: ^pandas/` - closing then, as I now think it's probably not worth it

**Pull request closed without merging**